rest-server/engines/src/main/scala/com/actionml/engines/ur/URAlgorithm.scala

```diff
@@ -496,13 +497,29 @@ class URAlgorithm private (
 
     val userHistBias = query.userBias.getOrElse(userBias)
     val userEventsBoost = if (userHistBias > 0 && userHistBias != 1) Some(userHistBias) else None
+    // create a Map of alias -> indicator name or indicator name -> indicator name if no aliases
+    val queryEventNames = indicatorParams.flatMap { case (indicatorName, iParams) =>
+      val aliases = iParams.aliases.getOrElse(Seq(indicatorName))
+      if(indicatorName == aliases.head) {
```

Comment:
I'd suggest `aliases.contains(indicatorName)` or `alicases.headOption.contains(indicatorName` (if first position matters) to avoid NPE